packages/core/src/invoke.ts

```diff
@@ -177,15 +209,20 @@ export function spawnCallback<TE extends EventObject = AnyEventObject>(
       parent.send(error(id, err) as any);
     }
 
-    // if (isPromiseLike(callbackStop)) {
-    //   // it turned out to be an async function, can't reliably check this before calling `callback`
-    //   // because transpiled async functions are not recognizable
-    //   return this.spawnPromise(callbackStop as Promise<any>, id);
-    // }
+    if (isPromiseLike(callbackStop)) {
+      // it turned out to be an async function, can't reliably check this before calling `callback`
+      // because transpiled async functions are not recognizable
+      return spawnPromise(callbackStop as Promise<any>)(ctx, event, {
+        parent,
+        id,
+        _event
+      });
+    }
 
     const actor = {
       id,
-      send: event => receivers.forEach(receiver => receiver(event)),
+      send: receivedEvent =>
```

Comment:
I find this a little bit odd, not sure right now what would be the ideal API - just raising my concern out loud. 

Given that we assume everything is an actor it maybe would be more appropriate to allow only for a single receiver on a callback's side. Just like actors have a single mailbox. Registering multiple receivers can only make control flow convoluted and I feel that is an easy way to make a mistake along the way - so if we could figure out an API that would avoid those pitfalls then it would be great.

Reply:
A proposal for such an API would be to expect `{send,stop}` return type here. Promises, observables and machines have onown interfaces so we know how to adapt them to the actor API but as callbacka are much more free-form we cannot do that automatically like for those other types. So maybe just require returning this clear pair instead of providing a custom way to integrate callbacks. It would nicely relate to all other places where actor interface is used

Reply:
`spawnCallback` will be removed in a future PR. This is just to get the tests to pass. You're right, though.